### ISSUE TITLE
Can not focus on TinyMCE editor when ILayoutAware is added to content type

### ISSUE THREAD

**Issue by neilferreira** [plone/plone.app.mosaic#305]
I am currently unable to use TinyMCE on to edit any RichText fields on my content item under the Properties tab.  Is anyone else affected by this same behaviour?

I can see items in the changelog that relate to this in 2.0.0rc1 from @vangheem but nothing seems really obvious to me at this time.

Any info would be appreciated

Versions:
```
plone.app.mosaic=2.0.0rc2
plone.app.blocks=4.0.0rc1
```

**Comment by datakurre** [comment]
@neilferreira I can confirm this. Here's a demo from the RC2 release:

http://plonemosaic200rc2.herokuapp.com/Plone/login

We probably didn't notice this, because TinyMCE looked like it initialized also on properties. Just didn't work.

**Comment by neilferreira** [comment]
Any idea where to start looking at fixing it? I'm on a tight deadline and
don't want to try backtracking too many versions to the last point where it
was stable. I'm happy to spend some time looking at it tomorrow morning.

**Comment by datakurre** [comment]
@neilferreira By trying again, I managed to reproduce this issue only with the first edit of a new content (when content layout menu appears and you need to select the first layout). On subsequent edits everything worked, and changing layout again did not break TinyMCE.

I believe, all that happens in mosaic.core.js.

TinyMCE is still alive when layout menu is visible.

**Comment by neilferreira** [comment]
Our workaround to have TinyMCE ready for our production build is to do the following:   Every time you need to make a change to any richtext field.  Swap the mimetype using the dropdown below the editor to  text/x-web-texttile and then back to text/html, this allows you to enter context.  If you still can't enter any context, try using the source editor and enter some dummy HTML then do the switching process again.

**Comment by datakurre** [comment]
The workaround works indeed. I can confirm that this is still an issue with Plone 5.1 (with the latest TinyMCE), so probably we need to figure out a fix either in mockup or in plone.app.mosaic.

What I remember is that Mosaic does not really touch DOM of the original fields – it just changes their CSS styles, so this may also be an issue of TinyMCE pattern mockup.

**Comment by datakurre** [comment]
I was not aware that with the current mockup overlay, the DOM is actually modified (that was not the case with original mosaic overlay). That modification could be the reason for dead TinyMCE, which historically has not survived being moved around DOM.

So, once overlay is created, all TinyMCEs must be re-initialized (destroyed before creating the modal and re-creating after the modal is built). I'm not sure, which part of this is currently done.

**Comment by datakurre** [comment]
Finally found a cure for this, but it will take some time before I get to merge and release this:

https://github.com/plone/plone.app.mosaic/commit/d4d0a75210a1846ff90b0c99cf71438f6bb38a19

I finally had time to look into the current special overlay used for properties tab. It was really so, that to make the overlay look more like the modal from mockup, a DOM move was added to the overlay. But DOM move breaks TinyMCE (it's supposedly related to iframes). I added code to destroy and re-init the pattern controlling TinyMCE (either pat-tinymce or pat-textareamimetypeselector) before and after the DOM move.
